contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+        if (!lct.verifyRegisterDerivativeForAll(derivativeIpId, originalIpIds, licenseTermsIds, derivativeIpOwner)) {
+            revert Errors.LicensingModule__LicenseTokenNotCompatibleForDerivative(derivativeIpId, licenseTokenIds);
+        }
+
+        LICENSE_REGISTRY.registerDerivativeIp(derivativeIpId, originalIpIds, licenseTemplate, licenseTermsIds);
+
+        address commonRoyaltyPolicy = address(0);
+        bytes[] memory royaltyDatas = new bytes[](originalIpIds.length);
+        for (uint256 i = 0; i < originalIpIds.length; i++) {
+            (address royaltyPolicy, bytes memory royaltyData, , ) = lct.getRoyaltyPolicy(licenseTermsIds[i]);
+            royaltyDatas[i] = royaltyData;
+            if (i == 0) {
+                commonRoyaltyPolicy = royaltyPolicy;
+            } else if (royaltyPolicy != commonRoyaltyPolicy) {
+                revert Errors.LicensingModule__IncompatibleRoyaltyPolicy(royaltyPolicy, commonRoyaltyPolicy);
+            }
+        }
+
+        if (commonRoyaltyPolicy != address(0)) {
+            ROYALTY_MODULE.onLinkToParents(
+                derivativeIpId,
+                commonRoyaltyPolicy,
+                originalIpIds,
+                royaltyDatas,
+                royaltyContext
+            );
+        }
+        // burn license tokens
+        LICENSE_NFT.burnLicenseTokens(derivativeIpOwner, licenseTokenIds);
+        emit DerivativeRegisteredWithLicenseTokens(
+            msg.sender,
+            derivativeIpId,
+            licenseTokenIds,
+            originalIpIds,
+            licenseTermsIds,
+            licenseTemplate
+        );
+    }
+
+    function _payMintingFeeForAll(
+        address[] calldata originalIpIds,
+        uint256[] calldata licenseTermsIds,
+        address licenseTemplate,
+        address derivativeIpOwner,
+        bytes calldata royaltyContext
+    ) private returns (address commonRoyaltyPolicy, bytes[] memory royaltyDatas) {
+        commonRoyaltyPolicy = address(0);
+        royaltyDatas = new bytes[](licenseTermsIds.length);
+
+        for (uint256 i = 0; i < originalIpIds.length; i++) {
+            uint256 lcId = licenseTermsIds[i];
+            Licensing.MintingLicenseSpec memory mlc = LICENSE_REGISTRY.getMintingLicenseSpec(
+                originalIpIds[i],
+                licenseTemplate,
+                lcId
+            );
+            if (mlc.receiverCheckModule != address(0)) {
+                if (!IHookModule(mlc.receiverCheckModule).verify(derivativeIpOwner, mlc.receiverCheckData)) {
+                    revert Errors.LicensingModule__ReceiverCheckFailed(derivativeIpOwner);
+                }
+            }
+            (address royaltyPolicy, bytes memory royaltyData) = _payMintingFee(
+                originalIpIds[i],
+                licenseTemplate,
+                lcId,
+                1,
+                royaltyContext,
+                mlc
+            );
+            royaltyDatas[i] = royaltyData;
+
+            if (i == 0) {
```

Comment:
what does it mean here i=0?  root IPA?  please add some comments.

Reply:
Yes, added more comments. it is mainly for checking the royaltyPolicy of all license terms must be the same.
the `i == 0` is for initializing value of commonRoyaltyPolicy with royalty policy of the first license.